src/gui/qgsrelationeditorwidget.cpp

```diff
@@ -189,18 +191,39 @@ QgsRelationEditorWidget::QgsRelationEditorWidget( const QVariantMap &config, QWi
   mViewModeButtonGroup = new QButtonGroup( this );
   mViewModeButtonGroup->addButton( mFormViewButton, QgsDualView::AttributeEditor );
   mViewModeButtonGroup->addButton( mTableViewButton, QgsDualView::AttributeTable );
+  // multiedit info label
+  mMultiEditInfoLabel = new QLabel( this );
+  buttonLayout->addWidget( mMultiEditInfoLabel );
 
   // add buttons layout
   rootLayout->addLayout( buttonLayout );
 
-  // add dual view
-  QGridLayout *relationLayout = new QGridLayout();
-  relationLayout->setContentsMargins( 0, 0, 0, 0 );
+  // add stacked widget
+  mStackedWidget = new QStackedWidget( this );
+
+  // add dual view (single feature content)
   mDualView = new QgsDualView( this );
   mDualView->setView( mViewMode );
   connect( mDualView, &QgsDualView::showContextMenuExternally, this, &QgsRelationEditorWidget::showContextMenu );
-  relationLayout->addWidget( mDualView );
-  rootLayout->addLayout( relationLayout );
+
+  // add multi feature editing page
+  mMultiEditStackedWidgetPage = new QWidget( this );
+  {
```

Comment:
Just out of curiosity, why are you starting a new block ? In case somebody need to define an other vboxlayout in the same method ?

Reply:
I like to let variables get out of scope as soon as they are not used anymore. In this case is not so important but if instead of a pointer it was for example a QPixmap object, by the closing bracket the object will be destroyed and the memory freed again.
It's an habit I took from programming embedded systems with limited resources, but if it doesn't fit in the QGIS coding guidelines I can remove the brackets :)

Reply:
I think it fits QGIS guidelines, you can let it that way